### PR TITLE
Fix No data selected appearance

### PR DESCRIPTION
**Pull request by AMoo-Miki**
Post https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8665

Fix No data selected appearance
Also:
* Fix some React errors

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/ui/query_editor/editors/default_editor/index.tsx [hunk 1]
```diff
@@ -77,14 +77,22 @@ export const DefaultInput: React.FC<DefaultInputProps> = ({
             gutterSize="none"
             className="defaultEditor__footerRow"
           >
-            {footerItems.start?.map((item) => (
-              <EuiFlexItem grow={false} className="defaultEditor__footerItem">
+            {footerItems.start?.map((item, idx) => (
+              <EuiFlexItem
+                key={`defaultEditor__footerItem-start-${idx}`}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/query_editor/editors/default_editor/index.tsx:82]: Just added key to get rid of react warning.

File: src/plugins/data/public/ui/query_editor/editors/default_editor/index.tsx [hunk 2]
```diff
@@ -77,14 +77,22 @@ export const DefaultInput: React.FC<DefaultInputProps> = ({
...
+              >
                 {item}
               </EuiFlexItem>
             ))}
             <EuiFlexItem grow />
-            {footerItems.end?.map((item) => (
-              <EuiFlexItem grow={false} className="defaultEditor__footerItem">
+            {footerItems.end?.map((item, idx) => (
+              <EuiFlexItem
+                key={`defaultEditor__footerItem-end-${idx}`}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/query_editor/editors/default_editor/index.tsx:92]: Just added key to get rid of react warning.

File: src/plugins/data/public/ui/filter_bar/filter_options.tsx [hunk 3]
```diff
@@ -88,8 +88,7 @@ const FilterOptionsUI = (props: Props) => {
   const uiSettings = opensearchDashboards.services.uiSettings;
   const isPinned = uiSettings!.get(UI_SETTINGS.FILTERS_PINNED_BY_DEFAULT);
   const useNewHeader = Boolean(uiSettings!.get(UI_SETTINGS.NEW_HOME_PAGE));
-  const [indexPattern] = props.indexPatterns;
-  const index = indexPattern && indexPattern.id;
+  const index = Array.isArray(props.indexPatterns) ? props.indexPatterns[0]?.id : undefined;
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/filter_bar/filter_options.tsx:91]: Needed to make the no-index-pattern state to pass through.

**Review by ashwin-pc — DISMISSED** [review]

File: src/plugins/data/public/ui/query_editor/language_selector.tsx [hunk 4]
```diff
@@ -111,7 +111,6 @@ export const QueryLanguageSelector = (props: QueryLanguageSelectorProps) => {
       button={
         <EuiSmallButtonEmpty
           iconSide="right"
-          iconSize="s"
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/query_editor/language_selector.tsx:114]: EuiSmallButtonEmpty doesn't handle it, even tough it claims to do.

File: src/plugins/discover/public/application/view_components/canvas/index.tsx [hunk 5]
```diff
@@ -143,41 +143,47 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history, optionalR
...
-          </>
-        )}
-        {fetchState.status === ResultStatus.READY && !isEnhancementsEnabled && (
-          <EuiPanel hasShadow={false} paddingSize="none" className="dscCanvas_results">
-            <MemoizedDiscoverChartContainer {...fetchState} />
-            <MemoizedDiscoverTable rows={rows} scrollToTop={scrollToTop} />
-          </EuiPanel>
-        )}
-      </>
+      {indexPattern ? (
```
Comment (AMoo-Miki) [inline src/plugins/discover/public/application/view_components/canvas/index.tsx:146]: Added the else to stop showing the spinner.

**Review by sejli — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Pull request merged**